### PR TITLE
Capture and parse the cmdstat timings from `INFO ALL` command

### PR DESCRIPTION
**Pull request by adamflott**
### Required for all PRs:

- [X] Signed [CLA](https://influxdata.com/community/cla/).
- [X] Associated README.md updated.
- [X] Has appropriate unit tests.

### CODE REVIEW

**Comment by glinton** [comment]
There is already an open PR for this in #5874. What do you feel this adds that that one doesn't provide?

**Comment by adamflott** [comment]
> There is already an open PR for this in #5874. What do you feel this adds that that one doesn't provide?

Oops! Didn't see that. The only difference is, their PR does 2 remote calls, and mine does 1. Either way, I don't care which one is merged, just that it is :)

**Comment by wingyplus** [comment]
@adamflott your pr seems to be make sense than my opened pr. :)

File: plugins/inputs/redis/redis.go [hunk 1]
```diff
@@ -321,6 +326,47 @@ func gatherKeyspaceLine(
...
+		fields := make(map[string]interface{})
+		tags := make(map[string]string)
+		for k, v := range global_tags {
+			tags[k] = v
+		}
+		tags["command"] = strings.TrimPrefix(name, "cmdstat_")
+		parts := strings.Split(line, ",")
+		for _, part := range parts {
+			kv := strings.Split(part, "=")
+			if len(kv) == 2 {
```
Comment (wingyplus) [inline plugins/inputs/redis/redis.go]: Invert if can reduce indentation

File: plugins/inputs/redis/redis_test.go [hunk 2]
```diff
@@ -116,6 +116,22 @@ func TestRedis_ParseMetrics(t *testing.T) {
...
+	}
+	acc.AssertContainsTaggedFields(t, "redis_cmdstat", cmdstatSetFields, cmdstatSetTags)
+
+	cmdstatCommandTags := map[string]string{"host": "redis.net", "replication_role": "master", "command": "command"}
+	cmdstatCommandFields := map[string]interface{}{
+		"calls":         int64(1),
+		"usec":          int64(990),
+		"usec_per_call": float64(990.0),
+	}
+	acc.AssertContainsTaggedFields(t, "redis_cmdstat", cmdstatCommandFields, cmdstatCommandTags)
```
Comment (wingyplus) [inline plugins/inputs/redis/redis_test.go]: Should we needs to add test in case that usec and usec_per_call are not number?

**Comment by adamflott** [comment]
@glinton any updates? I would like to see either #5874 or this PR get merged soon.

**Comment by glinton** [comment]
This likely won't make it into 1.12 as we're cramming trying to finish up what is already in the milestone. Because this is a basic change though, it is possible it gets into the 1.13 release

File: plugins/inputs/redis/redis.go (hunk 1 above)
Comment (glinton) [reply plugins/inputs/redis/redis.go]: agreed, `continue` `if len(kv) != 2`

File: plugins/inputs/redis/redis.go [hunk 3]
```diff
@@ -321,6 +326,47 @@ func gatherKeyspaceLine(
...
+// Example:
+//     cmdstat_publish:calls=33791,usec=208789,usec_per_call=6.18
+// Tag: cmdstat=publish; Fields: calls=33791i,usec=208789i,usec_per_call=6.18
+func gatherCommandstateLine(
+	name string,
+	line string,
+	acc telegraf.Accumulator,
+	global_tags map[string]string,
+) {
+	if strings.HasPrefix(name, "cmdstat") {
```
Comment (glinton) [inline plugins/inputs/redis/redis.go]: `return` `if !strings.HasPrefix`

File: plugins/inputs/redis/redis_test.go (hunk 2 above)
Comment (glinton) [reply plugins/inputs/redis/redis_test.go]: If this is something that may be returned, let's test for and handle it.

**Comment by glinton** [comment]
Resolves #2894 	
Preferred over and closes #2902 

@adamflott can you also document the additional fields this adds, [similar to 2902](https://github.com/influxdata/telegraf/pull/2902/files#diff-a112c896f52189329f25987037c5d3f3R88)

**Comment by glinton** [comment]
@adamflott not sure if you saw, but I was mistaken when i mentioned this would be pushed off until 1.13. Do you foresee yourself getting time to address the latest feedback?

**Comment by adamflott** [comment]
> @adamflott not sure if you saw, but I was mistaken when i mentioned this would be pushed off until 1.13. Do you foresee yourself getting time to address the latest feedback?

I can have this PR done by the end of tomorrow 8/14 EST.

I also have 2 other PRs around Redis I have vested interest in getting merged in

* https://github.com/influxdata/telegraf/pull/5978
* https://github.com/influxdata/telegraf/pull/5921

Any way I can get those reviewed?

File: plugins/inputs/redis/redis_test.go (hunk 2 above)
Comment (adamflott) [reply plugins/inputs/redis/redis_test.go]: Is there a pattern from testutil I could use?

FWIW, I don't expect the type to change given the age of this commit :)
https://github.com/antirez/redis/commit/d7ed7fd2fcd29fbba605fdd06b9498ffd85c092f

File: plugins/inputs/redis/redis.go (hunk 3 above)
Comment (adamflott) [reply plugins/inputs/redis/redis.go]: done

File: plugins/inputs/redis/redis.go [hunk 4]
```diff
@@ -321,6 +326,51 @@ func gatherKeyspaceLine(
...
+// Example:
+//     cmdstat_publish:calls=33791,usec=208789,usec_per_call=6.18
+// Tag: cmdstat=publish; Fields: calls=33791i,usec=208789i,usec_per_call=6.18
+func gatherCommandstateLine(
+	name string,
+	line string,
+	acc telegraf.Accumulator,
+	global_tags map[string]string,
+) {
+	if !strings.HasPrefix(name, "cmdstat") {
```
Comment (danielnelson) [inline plugins/inputs/redis/redis.go]: Nitpick, but I would add an underscore to match the TrimPrefix laster on:
```diff
- if !strings.HasPrefix(name, "cmdstat") {
+ if !strings.HasPrefix(name, "cmdstat_") {
```

**Comment by danielnelson** [comment]
The sentinel plugin will need to wait until 1.13, but the other should be able to make it.

**Pull request merged**